qiskit/assembler/assemble_schedules.py

```diff
@@ -170,15 +171,31 @@ def _assemble_instructions(
     acquire_instruction_map = defaultdict(list)
     for time, instruction in schedule.instructions:
 
-        if isinstance(instruction, ParametricInstruction):
+        if isinstance(instruction, Play) and isinstance(instruction.pulse, ParametricPulse):
+            pulse_shape = ParametricPulseShapes(type(instruction.pulse)).name
+            if pulse_shape not in run_config.parametric_pulses:
+                instruction = Play(instruction.pulse.get_sample_pulse(),
+                                   instruction.channel,
+                                   name=instruction.name)
+
+        if isinstance(instruction, ParametricInstruction):  # deprecated
             pulse_shape = ParametricPulseShapes(type(instruction.command)).name
             if pulse_shape not in run_config.parametric_pulses:
                 # Convert to SamplePulse if the backend does not support it
                 instruction = PulseInstruction(instruction.command.get_sample_pulse(),
                                                instruction.channels[0],
                                                name=instruction.name)
 
-        if isinstance(instruction, PulseInstruction):
+        if isinstance(instruction, Play) and isinstance(instruction.pulse, SamplePulse):
+            name = instruction.pulse.name
+            if instruction.pulse != user_pulselib.get(name):
```

Comment:
if we have two sample pulses with the identical samples and with different names, it seems like the assembler recognizes them as different pulses. is this intended? or do we combine them to save hardware resources?

Reply:
that's a good point. we should come back to this -- I'm going to say for now that it is out of scope for this PR (these lines duplicate existing behavior)